### ISSUE TITLE
Add an option/command to the formatter where declaration gets inserted on bindings

### ISSUE THREAD

**Issue by Marwes** [gluon-lang/gluon#436]
This is going to be a bit tricky since we need to decide which bindings should get types. It certainly can't be all bindings since that would mean annotating bindings such as `let x = 1` which is just noise.

I can think of three strategies here.

* Annotate all bindings that are exported from the module
```
// Annotate `let f x : Int -> Int ...`
let f x = x + 1
// Leave alone
let g y = y
{ f }
```

* Expand type holes
```
// Expand `_` into the actual type
let f x : _ = ...
```

* Insert the type on the binding under the cursor in the editor
```
let f x = ...
// ^ cursor here, insert `let f x : Type...`
```

**Comment by brendanzab** [comment]
I see this less as a formatting concern, and more of a interactive editing concern. Have you tried out interactive Agda and Idris sessions before? They can be super cool, with in-place case splits and hole expansion. You might also want to give a list of possible types if a hole could be more than one possibility.

**Comment by Marwes** [comment]
> Have you tried out interactive Agda and Idris sessions before

No, but it sounds like I'd want to try that before trying to implement stuff like this!

**Comment by brendanzab** [comment]
Lean is also a neat one to try, they have a really nice VS Code plugin. Alas they are lacking more on the code inference front - ie. case splitting and hole expansion.
